### DISCUSSION TITLE
How to add covariates by fine-tuning?

### DISCUSSION THREAD

**Discussion by Riofd** [amazon-science/chronos-forecasting#99] [Q&A]
Hello, I wan to know if there is any easy way to add covariates for forecasting by fine-tuning? Since Chronos does not support covariates, I just want to add more channels as input by adding a "cross-attention" layer at the input stage. My idea is to do this special "cross-attention" after input embedding. I want to freeze most of the layers and just train the new "cross-attention" layer and fine-tune the last linear layer (named `lm_head` in Chronos).

**Comment by Otsutsukii** [comment]
Interesting idea, maybe something I will try later, except if the authors came up with better ideas

**Reply by Riofd** [reply]
I think this is a easy way for downstream forecasting with covariates, and I am trying to do so. Looking forward to your feedback on the results~

**Comment by hexiaomingha** [comment]
How are your results?

**Comment by abdulfatir** [comment]
We just release Chronos-2 which natively supports covariates in a zero-shot manner. Check it out!
